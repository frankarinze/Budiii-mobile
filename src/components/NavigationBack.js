import React from 'react';
import { TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { gStyle } from '../constants';

// grab icon
import Icon from 'react-native-vector-icons/Ionicons';

const NavigationBack = ({ navigation }) => (
  <TouchableOpacity
    accessible
    accessibilityLabel="go back"
    accessibilityComponentType="button"
    accessibilityTraits="button"
    activeOpacity={gStyle.activeO}
    onPress={() => navigation.goBack(navigation.state.key)}
    style={{ paddingHorizontal: 16, paddingVertical: 8 }}
  >
   <Icon name="ios-arrow-back" size={20}  />
   
  </TouchableOpacity>
);

NavigationBack.propTypes = {
  // required
  navigation: PropTypes.object.isRequired
};

export default NavigationBack;
