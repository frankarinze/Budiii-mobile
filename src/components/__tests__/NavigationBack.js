import React from 'react';
import renderer from 'react-test-renderer';

import NavigationBack from '../NavigationBack';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn()
};

const tree = renderer
  .create(<NavigationBack navigation={navigation} />)
  .toJSON();

describe('<NavigationBack />', () => {
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('has 1 child', () => {
    expect(tree.children.length).toBe(1);
  });
});
