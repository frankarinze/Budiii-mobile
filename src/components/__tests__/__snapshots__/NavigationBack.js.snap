// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBack /> renders correctly 1`] = `
<View
  accessibilityLabel="go back"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={24}
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    >
      <RNSVGPath
        d="M16 32c8.837 0 16-7.163 16-16S24.837 0 16 0 0 7.163 0 16s7.163 16 16 16zm0-29c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3z"
        fill={
          Array [
            0,
            0.11372549019607843,
            0.17647058823529413,
            0.22745098039215686,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      />
      <RNSVGPath
        d="M20.914 9.914l-2.829-2.829-8.914 8.914 8.914 8.914 2.828-2.828-6.086-6.086z"
        fill={
          Array [
            0,
            0.11372549019607843,
            0.17647058823529413,
            0.22745098039215686,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
